pkg/controller/statusmanager/pod_status.go

```diff
@@ -57,23 +59,31 @@ type deploymentState struct {
 	LastChangeTime time.Time
 }
 
+// statefulsetState is the same as daemonsetState.. but for statefulsets!
+type statefulsetState struct {
+	types.NamespacedName
+
+	LastSeenStatus appsv1.StatefulSetStatus
+	LastChangeTime time.Time
+}
+
 // SetFromPods sets the operator Degraded/Progressing/Available status, based on
-// the current status of the manager's DaemonSets and Deployments.
+// the current status of the manager's DaemonSets, Deployments and StatefulSets.
 func (status *StatusManager) SetFromPods() {
 	status.Lock()
 	defer status.Unlock()
 
 	targetLevel := os.Getenv("RELEASE_VERSION")
-	reachedAvailableLevel := (len(status.daemonSets) + len(status.deployments)) > 0
+	reachedAvailableLevel := (len(status.daemonSets) + len(status.deployments) + len(status.statefulSets)) > 0
 
 	progressing := []string{}
 	hung := []string{}
 
-	daemonsetStates, deploymentStates := status.getLastPodState()
+	daemonsetStates, deploymentStates, statefulsetStates := status.getLastPodState()
 
 	for _, dsName := range status.daemonSets {
 		ds := &appsv1.DaemonSet{}
-		if err := status.client.Get(context.TODO(), dsName, ds); err != nil {
+		if err := status.client.Default().CRClient().Get(context.TODO(), dsName, ds); err != nil {
```

Comment:
Can we use `status.client.ClientFor(...` instead? if `clusterName` is not set it will pick the default client

Reply:
Done